bokehjs/src/coffee/models/axes/mercator_axis.ts

```diff
@@ -0,0 +1,42 @@
+import {AxisView} from "./axis"
+import {LinearAxis} from "./linear_axis"
+import {MercatorTickFormatter} from "../formatters/mercator_tick_formatter"
+import {MercatorTicker} from "../tickers/mercator_ticker"
+
+export class MercatorAxisView extends AxisView {
+  model: MercatorAxis
+}
+
+export namespace MercatorAxis {
+  export interface Attrs extends LinearAxis.Attrs {
+    ticker:    MercatorTicker
+    formatter: MercatorTickFormatter
+  }
+
+  export interface Props extends LinearAxis.Props {}
+}
+
+export interface MercatorAxis extends MercatorAxis.Attrs {}
+
+export class MercatorAxis extends LinearAxis {
+
+  properties: MercatorAxis.Props
+
+  ticker:    MercatorTicker
+  formatter: MercatorTickFormatter
```

Comment:
This shouldn't be necessary. `export interface MercatorAxis` makes sure that `Attrs` are both part of the interface and are "implemented". There are a few rare cases where this doesn't wok though.

Reply:
Doesn't this specialize them to be required to be mercator ticker and formatter specifically?

Reply:
Yes, but you already did this in `Attrs` and by using `export interface MercatorAxis extends MercatorAxis.Attrs {}`. Those two lines would be necessary, if `export interface MercatorAxis (...)` wasn't there. This is a way to avoid duplication.

Reply:
Also, you don't get any errors, because what you do is a valid (yet trivial) override of already defined members.

Reply:
If i remove those lines VSCode reports an error:

> file: 'file:///Users/bryanv/work/bokeh/bokehjs/src/coffee/models/axes/mercator_axis.ts'
severity: 'Error'
message: 'Interface 'MercatorAxis' cannot simultaneously extend types 'LinearAxis' and 'Attrs'.
  Named property 'formatter' of types 'LinearAxis' and 'Attrs' are not identical.'
at: '19,18'
source: 'ts'
code: '2320'

Reply:
did you also mean remove the lines in `Attrs` as well?

Reply:
I see, so this is the one case where this is necessary. Just leave it as it is now.

Reply:
OK I rebased to removed the commit